### DISCUSSION TITLE
how select the area between two notes

### DISCUSSION THREAD

**Discussion by 262146150** [opensheetmusicdisplay/opensheetmusicdisplay#1575] [Q&A] [answered]
If you want to select the area between two notes, how do you do it? For example, by clicking on the first starting note and the second ending note,thanks

**Comment by sschmidTU** [comment]
What do you mean? You want to display a rectangle between them? Can you add an image that illustrates this?

**Reply by 262146150** [reply]
Thank you for your reply. What I mean is that we allow students to practice a specific section of the entire score. What we hope to achieve is that after loading the score, they can select the starting and ending notes, and there will be a highlighted display of the area between the start and end.

**Reply by sschmidTU** [reply, marked as answer]
That's your job to implement, OSMD is a sheet music renderer, not a tool that does everything for you out of the box.
You can take the cursor as inspiration and simply draw between the end and start x of the cursor positions instead of what the cursor does.
<img width="69" alt="image" src="https://github.com/user-attachments/assets/3aeaf4b9-a03e-4bf7-a902-b51fd5d28d03">

Also see this:
https://github.com/opensheetmusicdisplay/opensheetmusicdisplay/wiki/Exploring-the-Demo#drawing-overlay-lines-over-the-score-and-getting-a-notes-position

**Reply by sschmidTU** [reply]
Here's the relevant cursor function update():
https://github.com/opensheetmusicdisplay/opensheetmusicdisplay/blob/aae428e6b880d08b65e21822a635525049c8d204/src/OpenSheetMusicDisplay/Cursor.ts#L134
You should be able to implement it similarly.

**Reply by 262146150** [reply]
thanks，I found a similar image that can illustrate my needs. It involves determining the start and end by clicking on the notes, then highlighting the area, and it can be at any position. Could I get some ideas on how to achieve this?
![WechatIMG23](https://github.com/user-attachments/assets/0e6a7e38-9744-48c9-8cae-1c6fade7320c)

**Reply by sschmidTU** [reply]
I just replied how you can achieve this, see above.
